documentation/book/assembly-upgrade-resources.adoc

```diff
@@ -8,8 +8,23 @@ For this release of {ProductName}, resources that use the API version `{KafkaApi
 
 The `{KafkaApiVersionPrev}` API version is deprecated in release {ProductVersion}.
 
+This section describes the upgrade steps for the resources.
+
 IMPORTANT: The upgrade of resources _must_ be performed after xref:assembly-upgrade-{context}[upgrading the Cluster Operator], so the Cluster Operator can understand the resources.
 
+.What if the resource upgrade does not take effect?
+If the upgrade does not take effect and a warning is given on reconciliation, make a cosmetic change to the custom resource, such as adding an annotation, to trigger the update.
```

Comment:
I think we should be more specific about the warning (and be clear that it's a warning in the _logs_). The warning message is like

"Cannot update status of resource Kafka named _resource-name_. The resource needs to be updated to newer apiVersion first."